src/back/msl/writer.rs

```diff
@@ -1437,10 +1475,49 @@ impl<W: Write> Writer<W> {
         writeln!(self.out, "#include <simd/simd.h>")?;
         writeln!(self.out)?;
 
+        let uses_unsized_buffers = {
```

Comment:
Instead of having `uses_unsized_buffers`, we can just check `if !self.runtime_sized_buffers.is_empty()` in the other places.
In addition, we should be clearing `runtime_sized_buffers` here, for the case where the same writer is re-used for different modules.